src/Polysemy/RandomFu.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-|
+Module      : PolySemy.RandomFu
+Description : Polysemy random-fu effect
+
+Polysemy "random-fu" effect.
+This can be run in a few ways:
+1. Directly in 'IO'
+2. Using any 'Data.Random.RandomSource' from "random-fu"
+3. In 'IO', using a given 'Data.Random.Source.PureMT' source. ('IO' is used to put the source in an 'IORef')
+
+
+Also, its presence in the Polysemy Union @r@,
+gives @Sem r@ a "MonadRandom" instance (from "random-fu"). This is an Orphan instance.
+-}
+
+module Polysemy.RandomFu
+  (
+    -- * Effect
+    Random (..)
+
+    -- * Actions
+  , sampleRVar
+  , getRandomPrim
+  , sampleDist
+
+  -- * Interpretations
+  , runRandomSource
+  , runRandomIO
+  , runRandomIOPureMT
+  )
+where
+
+import           Polysemy                      
+
+import           Data.IORef                     ( newIORef )
+import qualified Data.Random                   as R
+import qualified Data.Random.Source            as R
+import           Data.Random.Source.Std         (StdRandom(..))
+import qualified Data.Random.Internal.Source   as R
+import qualified Data.Random.Source.PureMT     as R
+import qualified System.Random                 as SR
+import           Control.Monad.IO.Class         ( MonadIO(..) )
+
+------------------------------------------------------------------------------
+{- | An effect capable of sampling from a "random-fu" RVar or generating a
+single random-variate of any type, @t@ with a
+@Data.Random.Prim t@ constructor, currently one of @Word8@, @Word16@,
+@Word32@, @Word64@, @Double@ or N-byte integer.
+-}
+data Random m r where
```

Comment:
RandomFu

Reply:
I'm happy to make this change but it's hard to imagine circumstances where a user would want both and the name clashes would be an issue.

Reply:
Yeah, I hear you. I think it's a good idea for a few reasons, though:

1. Reinforce the idea that the effect `E` comes from module `Polysemy.E`
2. Unique names let you disambiguate effects at a glance, rather than needing to look at which `Random` effect is being imported. Especially since they have very different APIs.